discovery/gossiper.go

```diff
@@ -3593,3 +3667,180 @@ func (d *AuthenticatedGossiper) ShouldDisconnect(pubkey *btcec.PublicKey) (
 
 	return false, nil
 }
+
+// validateFundingTransaction fetches the channel announcements claimed funding
+// transaction from chain to ensure that it exists, is not spent and matches
+// the channel announcement proof. The transaction's outpoint and value are
+// returned if we can glean them from the work done in this method.
+func (d *AuthenticatedGossiper) validateFundingTransaction(
+	ann *lnwire.ChannelAnnouncement1,
+	tapscriptRoot fn.Option[chainhash.Hash]) (fn.Option[wire.OutPoint],
+	fn.Option[btcutil.Amount], error) {
```

Comment:
don't think they need to be options?

Reply:
i think the og reason was case I had the `if AssumeValid { return nil, nil }` in here meaning we'd have a nil error but also nil return values. But in the latest iteration ive moved that check outside of the validate method (ie, now we only call `validate` if !AssumeValid.